core/node/state_keeper/src/batch_executor/main_executor.rs

```diff
@@ -68,21 +62,17 @@ impl BatchExecutor for MainBatchExecutor {
             commands: commands_receiver,
         };
 
-        let stop_receiver = stop_receiver.clone();
         let handle = tokio::task::spawn_blocking(move || {
-            if let Some(storage) = Handle::current()
-                .block_on(
-                    storage_factory.access_storage(&stop_receiver, l1_batch_params.number - 1),
-                )
-                .context("failed accessing state keeper storage")?
-            {
-                executor.run(storage, l1_batch_params, system_env);
-            } else {
-                tracing::info!("Interrupted while trying to access state keeper storage");
-            }
+            let storage = match storage {
+                OwnedStorage::Static(storage) => storage,
+                OwnedStorage::Lending(ref storage) => Handle::current()
```

Comment:
JFYI: This is one of the few scenarios in which `ref` is seemingly required: the branch above moves `storage`, and in this branch, we must borrow from it.